packages/compartment-mapper/src/assemble.js

```diff
@@ -2,15 +2,110 @@
 /// <reference types="ses" />
 
 import { resolve } from './node-module-specifier.js';
-import { mapParsers } from './parse.js';
+import { parseExtension } from './extension.js';
 
-const { entries, freeze } = Object;
+const { entries, fromEntries, freeze } = Object;
 
 const defaultCompartment = Compartment;
 
 // q, as in quote, for strings in error messages.
 const q = JSON.stringify;
 
+/**
+ * @template {string | number} Key
+ * @template Value
+ * @param {Record<Key, Value>} object
+ * @param {Key} key
+ * @returns {boolean}
+ */
+const has = (object, key) => Object.prototype.hasOwnProperty.call(object, key);
+
+/**
+ * @param {Record<string, string>} languageForExtension - maps a file extension
+ * to the corresponding language.
+ * @param {Record<string, string>} languageForModuleSpecifier - In a rare case,
+ * the type of a module is implied by package.json and should not be inferred
+ * from its extension.
+ * @param {Record<string, ParseFn>} parserForLanguage
+ * @param {ModuleTransforms} transforms
+ * @returns {ParseFn}
+ */
+export const makeExtensionParser = (
```

Comment:
Do you expect to use this outside the module? I ask because it's exported but current only used below, and because of my comment on line 42.